src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/CalculationController.kt

```diff
@@ -449,6 +454,35 @@ class CalculationController(
     return result.getOrElse { problemMessage -> throw CrdWebException(problemMessage, HttpStatus.NOT_FOUND) }
   }
 
+  @GetMapping(value = ["/nomis-calculation-summary/{offenderSentCalculationId}"])
+  @PreAuthorize("hasAnyRole('SYSTEM_USER', 'RELEASE_DATES_CALCULATOR')")
+  @ResponseBody
+  @Operation(
+    summary = "Get Nomis calculation summary with release dates for a offenderSentCalculationId",
+    description = "This endpoint will return the nomis calculation summary with release dates based on a offenderSentCalculationId",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "200", description = "Returns Nomis calculation summary with release dates based on a offenderSentCalculationId"),
+      ApiResponse(responseCode = "401", description = "Unauthorised, requires a valid Oauth2 token"),
+      ApiResponse(responseCode = "403", description = "Forbidden, requires an appropriate role"),
+      ApiResponse(responseCode = "404", description = "No nomis calculation summary - release dates exists for this offenderSentCalculationId"),
+    ],
+  )
+  fun getNomisCalculationSummary(
+    @Parameter(required = true, example = "123456", description = "The offenderSentCalculationId of the offender booking or a calculation")
+    @PathVariable("offenderSentCalculationId")
+    offenderSentCalculationId: Long,
+  ): NomisCalculationSummary {
+    log.info("Request received to get offender key dates for $offenderSentCalculationId")
+    val nomisOffenderKeyDates = prisonService.getNOMISOffenderKeyDates(offenderSentCalculationId)
+      .getOrElse { problemMessage -> throw CrdWebException(problemMessage, HttpStatus.NOT_FOUND) }
+    val releaseDatesForSentCalculationId = latestCalculationService.releaseDates(nomisOffenderKeyDates)
+    val detailsReleaseDates = calculationResultEnrichmentService.addDetailToCalculationDates(releaseDatesForSentCalculationId, null, null).values.toList()
+    val nomisReason = prisonService.getNOMISCalcReasons().find { it.code == nomisOffenderKeyDates.reasonCode }?.description ?: nomisOffenderKeyDates.reasonCode
```

Comment:
I don't think there is a test for default NOMIS reason

Reply:
Added - 'Test NomisCalculationSummary not existing code reason code'  test case in [OffenderKeyDatesServiceTest.kt] to cover the requested scenario.